server/app/services/geo/esri/EsriServiceAreaValidationConfig.java

```diff
@@ -0,0 +1,124 @@
+  public Optional<ConfigList> ESRI_ADDRESS_SERVICE_AREA_VALIDATION_PATHS;
+
+  @Inject
+  public EsriServiceAreaValidationConfig(Config configuration) {
+    this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_LABELS =
+        configuration.hasPath("esri_address_service_area_validation_labels")
+            ? Optional.of(configuration.getList("esri_address_service_area_validation_labels"))
+            : Optional.empty();
+    this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_VALUES =
+        configuration.hasPath("esri_address_service_area_validation_values")
+            ? Optional.of(configuration.getList("esri_address_service_area_validation_values"))
+            : Optional.empty();
+    this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_URLS =
+        configuration.hasPath("esri_address_service_area_validation_urls")
+            ? Optional.of(configuration.getList("esri_address_service_area_validation_urls"))
+            : Optional.empty();
+    this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_PATHS =
+        configuration.hasPath("esri_address_service_area_validation_paths")
+            ? Optional.of(configuration.getList("esri_address_service_area_validation_paths"))
+            : Optional.empty();
+  }
+
+  /** Checks if each element necessary for Esri address service area validation is present. */
+  public Boolean hasAllElements() {
+    if (this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_LABELS.isEmpty()
+        || this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_VALUES.isEmpty()
+        || this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_URLS.isEmpty()
+        || this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_PATHS.isEmpty()) {
+      return false;
+    }
+
+    return true;
+  }
+
+  private Optional<ImmutableList<String>> getConfigListSetting(Optional<ConfigList> setting) {
+    if (setting.isEmpty()) {
+      return Optional.empty();
+    }
+
+    ImmutableList.Builder<String> listBuilder = ImmutableList.builder();
+    setting.get().stream()
+        .forEach(configValue -> listBuilder.add((String) configValue.unwrapped()));
+    return Optional.of(listBuilder.build());
+  }
+
+  /** Returns the labels specified for Esri address service area validation. */
+  public Optional<ImmutableList<String>> getLabels() {
+    return getConfigListSetting(this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_LABELS);
+  }
+
+  /** Returns the values specified for Esri address service area validation. */
+  public Optional<ImmutableList<String>> getValues() {
+    return getConfigListSetting(this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_VALUES);
+  }
+
+  /** Returns the urls specified for Esri address service area validation. */
+  public Optional<ImmutableList<String>> getUrls() {
+    return getConfigListSetting(this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_URLS);
+  }
+
+  /** Returns the paths specified for Esri address service area validation. */
+  public Optional<ImmutableList<String>> getPaths() {
+    return getConfigListSetting(this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_PATHS);
+  }
```

Comment:
**suggestion (blocking):** none of these are consumed outside of the class, recommend removing

Reply:
removed these methods